Update Developer Guide:

jaya-r3 opened the pull request:
Resolve issue #9 

- Complete use cases for v1.2
- Update non-functional requirements

jaya-r3 committed:
Update Developer Guide:

- Complete use cases for v1.2
- Update non-functional requirements

sc-arecrow commented on docs/DeveloperGuide.md:
```diff
@@ -300,20 +301,48 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Glossary
 
-* **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **Mainstream OS**: Windows, Linux, Unix, macO
```
looks like a typo, should be macOS

sc-arecrow reviewed (changes requested):
LGTM other than a small typo

jaya-r3 committed:
Developer Guide:

- Fix typo relating to previous commit
- Edit Value Proposition to reflect v1.2 features

jaya-r3 committed:
Add EOF newline to DeveloperGuide.md

sc-arecrow reviewed (approved):
LGTM
